sql/postgresql/pggrammar.g4

```diff
@@ -0,0 +1,4294 @@
+    }
+
+        public static string unquote(string s)
+        {
+            var r = new StringBuilder(s.Length);
+            var i = 0;
+			while(i<s.Length)
+            {
+                var c = s[i];
+                r.Append(c);
+				if(c=='\''&&i<s.Length-1&&(s[i+1]=='\'')) i++;
+                i++;
+            }
+		return r.ToString();
+        }
+	        public static string GetRoutineBodyString(SconstContext rule)
+	        {
+	            var anysconst = rule.anysconst();
+	            var StringConstant = anysconst.StringConstant();
+	            if (null != StringConstant) return unquote(TrimQuotes(StringConstant.GetText()));
+	            var UnicodeEscapeStringConstant = anysconst.UnicodeEscapeStringConstant();
+	            if (null != UnicodeEscapeStringConstant) return TrimQuotes(UnicodeEscapeStringConstant.GetText());
+	            var EscapeStringConstant = anysconst.EscapeStringConstant();
+	            if (null != EscapeStringConstant) return TrimQuotes(EscapeStringConstant.GetText());
+	            string result = "";
+	            var dollartext = anysconst.DollarText();
+	            foreach (var s in dollartext)
+	            {
+	                result += s;
+	            }
+	            return result;
+	        }
+
+	    public static pggrammar getpggrammar(string script)
+	    {
+	       var CharStream = CharStreams.fromString(script);
+	       var Lexer = new pglexer(CharStream);
+	       var Tokens = new CommonTokenStream(Lexer);
+	       var Parser = new pggrammar(Tokens);
+	       var ErrorListener = new pggrammarErrorListener();
+	       ErrorListener.grammar = Parser;
+	       Parser.AddErrorListener(ErrorListener);
+	       return Parser;
+	    }
+
+	    internal class pggrammarErrorListener : BaseErrorListener
+        {
+            internal pggrammar grammar;
+	        public pggrammarErrorListener()
+	        {
+	        }
+	        public override void SyntaxError(TextWriter output, IRecognizer recognizer, IToken offendingSymbol, int line, int charPositionInLine, string msg, RecognitionException e)
+	        {
+	            grammar?.ParseErrors.Add(new ParseError(0, 0, line, charPositionInLine, msg));
+	        }
+	    }
+
+    public class ParseError
+    {
+        public ParseError(int number, int offset, int line, int column, string message)
+        {
+            Number = number;
+            Offset = offset;
+            Message = message;
+            Line = line;
+            Column = column;
+        }
+        public int Number { get; }
+        public int Offset { get; }
+        public int Line { get; }
+        public int Column { get; }
+        public string Message { get; }
+    }
+        private readonly IList<ParseError> m_ParseErrors = new List<ParseError>();
+        public IList<ParseError> ParseErrors => m_ParseErrors;
+        public IParseTree Root => root();
+        public IParseTree PlSqlRoot => plsqlroot();
+}
+
+root:
+	stmtblock EOF
+	;
```

Comment:
Could you please use the following format for all parser rules?

```g4
rule
    : alternative1
    | alternative2
    ...
    | alternativen
    ;
```

```suggestion
root
	: stmtblock EOF
	;
```

Reply:
@KvanTTT what plugin/tool can you advise for grammar reformating? Currently, I use Intellij Plugin for ANTLR 4.

Reply:
Unfortunately, I don't for now. Ok, let it be as is. Just pay attention to our grammar formatting rules for the future.